modules/compatibility/src/test/java/org/apache/ignite/compatibility/clients/ClientPartitionAwarenessMapperAPITestWrapper.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compatibility.clients;
+
+import java.io.Serializable;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.cache.affinity.AffinityFunction;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.client.ClientCache;
+import org.apache.ignite.client.ClientFeatureNotSupportedByServerException;
+import org.apache.ignite.client.ClientPartitionAwarenessMapper;
+import org.apache.ignite.client.ClientPartitionAwarenessMapperFactory;
+import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.configuration.ClientConfiguration;
+import org.apache.ignite.internal.util.typedef.X;
+import org.apache.ignite.lang.IgniteProductVersion;
+import static org.apache.ignite.compatibility.clients.JavaThinCompatibilityTest.ADDR;
+import static org.apache.ignite.compatibility.clients.JavaThinCompatibilityTest.CACHE_WITH_CUSTOM_AFFINITY;
+import static org.apache.ignite.internal.client.thin.ProtocolBitmaskFeature.ALL_AFFINITY_MAPPINGS;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrowsWithCause;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * This class is used to test a new API for partition awareness mapper factory added since 2.14 release.
+ *
+ * This wrapper is required to solve serialization/deserialization issues when the
+ * {@link JavaThinCompatibilityTest#testClient(IgniteProductVersion, IgniteProductVersion)} is used upon previous
+ * Ignite releases. The newly added classes must not be loaded by default when the test method is deserialized.
+ */
+public class ClientPartitionAwarenessMapperAPITestWrapper implements Serializable {
```

Comment:
Still don't understand why do we need this wrapper. Classes are loaded when method is invoked, for example `testServiceDescriptors` method contains `ClientServiceDescriptor` class, which was introduced in 2.13.